src/EditorFeatures/Test/Preview/PreviewWorkspaceTests.cs

```diff
@@ -238,6 +238,19 @@ public void TestPreviewWorkspaceDoesNotLeakSolution()
             solutionObjectReference.AssertReleased();
         }
 
+        [Fact]
+        public void TestPreviewWorkspaceDoesNotLeakItself()
+        {
+            var composition = EditorTestCompositions.EditorFeatures;
+            var exportProvider = composition.ExportProviderFactory.CreateExportProvider();
+            var previewWorkspaceReference = ObjectReference.CreateFromFactory<Workspace, TestComposition>(
+                static composition => new PreviewWorkspace(composition.GetHostServices()),
```

Comment:
Was there a reason the static lambda was necessary here? We don't care about allocations, was this needed for correctness?

Reply:
I wanted to make sure there was no implicit state machine or capture created for this method, so I know where all locals live.